### PR TITLE
feat: Add OnPush change detection

### PR DESCRIPTION
**Pull request by ert78gb**

### CODE REVIEW

File: src/app/day-calendar/day-calendar.component.ts [hunk 1]
```diff
@@ -130,6 +131,7 @@ export class DayCalendarComponent implements OnInit, OnChanges, ControlValueAcce
   }
 
   writeValue(value: CalendarValue): void {
+    console.log('inputValue equal: ', this.inputValue === value);
```
Comment (vlio20) [inline src/app/day-calendar/day-calendar.component.ts]: pls remove

Comment (ert78gb) [reply src/app/day-calendar/day-calendar.component.ts]: oh sry, removed

**Comment by ert78gb** [comment]
I will not use it without OnPush, because I use OnPush in the whole application and I don't wanna lose any performance.

**Comment by vlio20** [comment]
The question is, is there a workaround? because currently it makes it harder to develop new features to this component.

**Comment by ert78gb** [comment]
I don't know any workaround. I don't think harder to develop. Read https://angular-2-training-book.rangle.io/handout/change-detection/change_detection_strategy_onpush.html.

**Comment by vlio20** [comment]
I see there is a lot of over head for implementing new features. Maybe in the future when I will feel that there are no more features that I would like to add, then I will add this capability. Sorry for the time you wasted on implementing this feature - I learned from it a lot.

**Comment by ert78gb** [comment]
What kind of over head do you see?

**Comment by vlio20** [comment]
adding `cd.markForCheck();`. When I am not sure that I know exactly where to put it, and even if I will know, it would be harder to others contribute.

**Pull request closed without merging**